### ISSUE TITLE
Allow users to specify a `mongo` database name

### ISSUE THREAD

**Issue by doctorpangloss** [testcontainers/testcontainers-java#2876]
https://github.com/testcontainers/testcontainers-java/blob/556659751849dfb06313f46a6560d5a194efd035/modules/mongodb/src/main/java/org/testcontainers/containers/MongoDBContainer.java#L22

**Comment by pivovarit** [comment]
Why do you need it?

You can just access/create a custom instance using `MongoClient#getDatabase()`

**Comment by silaev** [comment]
When one uses Testcontainer’s reuse feature, a framework and a database per a test as a way to separate(isolate) data, then specifying a database might be helpful.
For instance, in the Spring Data MongoDB a database name is bind to the MongoTemplate/
ReactiveMongoTemplate and thus it’s a tedious process to change a database name for above-mentioned testing scenario. 

Given that I’ll submit a PR to add a withDatabaseName method for the MongoDBContainer. If the  withDatabaseName mothod is not called, then a database name will default to "test" as now.

**Comment by silaev** [comment]
Updated a corresponding [PR](https://github.com/testcontainers/testcontainers-java/pull/2980)

**Comment by PauloPortfolio** [comment]
@silaev How Can I solve this problem? In order to isolates each test from my test-class (my goal is to use PArallel tests, but for that I need isolates each test). Consequently, I need a different database-name in each test. For that, I need to change the TestContainer-Databasename. I am trying to use getReplicaSetUrl(String databaseName), but it is not working. Ho can I solve this problem?

**Comment by silaev** [comment]
> @silaev How Can I solve this problem? In order to isolates each test from my test-class (my goal is to use PArallel tests, but for that I need isolates each test). Consequently, I need a different database-name in each test. For that, I need to change the TestContainer-Databasename. I am trying to use getReplicaSetUrl(String databaseName), but it is not working. How can I solve this problem?

@PauloPortfolio Do you use a framework? Separation per the method of the class might not be so easy depending on a framework.

The key idea is to provide a database name to a MongoDB client and enable the Testcontainers' reuse feature.  Here is an example for the Testcontainers' MongoDB Module and Spring Data MongoDB to separate test classes (not methods!) per database names:

1. Enable reuse in .testcontainers.properties files by adding testcontainers.reuse.enable=true
2. Create a test class, for example:
```java
@DataMongoTest
@ContextConfiguration(initializers = IT1Test.Initializer.class)
class IT1Test {
  private static final MongoDBContainer MONGO_DB_CONTAINER =
    new MongoDBContainer(DockerImageName.parse("mongo:4.4.2"))
      .withReuse(true);
  @Autowired
  private ProductDao productDao;

  @BeforeAll
  static void setUpAll() {
    MONGO_DB_CONTAINER.start();
  }

  @AfterAll
  static void tearDownAll() {
    if (!MONGO_DB_CONTAINER.isShouldBeReused()) {
      MONGO_DB_CONTAINER.stop();
    }
  }

  @Test
  void test1() {
    Product product1 = ProductTestUtil.generateProduct(120589L, "AAA", Brand.DOLCE, BigDecimal.valueOf(9), 9, Size.SIZE_50);
    Product product2 = ProductTestUtil.generateProduct(120590L, "BBB", Brand.DOLCE, BigDecimal.valueOf(15.69), 6, Size.SIZE_100);
    Product product3 = ProductTestUtil.generateProduct(120591L, "CCC", Brand.ENGLISH_LAUNDRY, BigDecimal.valueOf(55.12), 3, Size.SIZE_100);
    StepVerifier.create(productDao.insert(Flux.just(product1, product2, product3)))
      .expectNextCount(3L)
      .verifyComplete();
  }

  static class Initializer implements ApplicationContextInitializer<ConfigurableApplicationContext> {
    @Override
    public void initialize(@NotNull ConfigurableApplicationContext configurableApplicationContext) {
      TestPropertyValues.of(
        String.format(
          "spring.data.mongodb.uri: %s",
          MONGO_DB_CONTAINER.getReplicaSetUrl("my-db" + UUID.randomUUID().getMostSignificantBits())
        )
      ).applyTo(configurableApplicationContext);
    }
  }
}
```

3. Run test 4 times 
4. Expect 4 new databases via the following mongo Shell command: db.getMongo().getDBNames(), example:
```json
[
        "admin",
        "config",
        "local",
        "my-db-5289724188915119626",
        "my-db-6086375598922775794",
        "my-db-79157823489553256",
        "my-db5367856646571576153"
]
```
5. Expect collections, for example via the following mongo Shell command: db.getMongo().getDB("my-db-5289724188915119626").product.count()

**Comment by eddumelendez** [comment]
Closing this issue due to it was implemented via #2980
